src/main/java/ch/fhnw/p2p/controller/ProjectController.java

```diff
@@ -0,0 +1,146 @@
+package ch.fhnw.p2p.controller;
+
+import ch.fhnw.p2p.authorization.AccessControl;
+import ch.fhnw.p2p.entities.Member;
+import ch.fhnw.p2p.entities.Project;
+import ch.fhnw.p2p.entities.Role;
+import ch.fhnw.p2p.entities.User;
+import ch.fhnw.p2p.repositories.MemberRepository;
+import ch.fhnw.p2p.repositories.ProjectRepository;
+import ch.fhnw.p2p.repositories.ProjectRepositoryImpl;
+import ch.fhnw.p2p.repositories.UserRepository;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.*;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.Valid;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * REST api controller for the project list
+ *
+ * @author Michelle Andrey
+ */
+
+@RestController
+@CrossOrigin(origins = "http://localhost:3000")
+@RequestMapping("/api/projects")
+public class ProjectController {
+	// ------------------------
+	// PRIVATE FIELDS
+	// ------------------------
+	private Log logger = LogFactory.getLog(this.getClass());
+
+	@Autowired
+	private AccessControl accessControl;
+
+	@Autowired
+	private ProjectRepository projectRepo;
+
+	@Autowired
+	UserRepository userRepo;
+	
+	@PersistenceContext
+	EntityManager entityManager;
+
+	// ------------------------
+	// PUBLIC METHODS
+	// ------------------------
+	
+	/**
+	 * Fetch full project (for editing).
+	 */
+	@RequestMapping(value="{id}", method = RequestMethod.GET)
+	public ResponseEntity<Project> getProject(HttpServletRequest request, @PathVariable Long id) {
+		User user = accessControl.login(request, AccessControl.Allowed.COACH);
+		logger.info("GET project user=" + user.getEmail());
+		Project project = projectRepo.findById(id);
+		return new ResponseEntity<Project>(project, HttpStatus.OK);
+	}
+
+	/**
+	 * Update an existing project.
+	 */
+	@RequestMapping(value="{id}", method = RequestMethod.PUT)
+	public ResponseEntity<HttpStatus> updateProject(HttpServletRequest request, @PathVariable Long id, @Valid @RequestBody Project project, BindingResult result) {
+		User user = accessControl.login(request, AccessControl.Allowed.COACH);
+		logger.info("PUT project user=" + user.getEmail());
+		if (result.hasErrors()) {
+			return new ResponseEntity<HttpStatus>(HttpStatus.PRECONDITION_FAILED);
+		}
+		Project oldProject = projectRepo.findById(project.getId());
+		if (oldProject == null) {
+			return new ResponseEntity<HttpStatus>(HttpStatus.NOT_FOUND);
+		}
+		oldProject.setTitle(project.getTitle());
```

Comment:
gut so! 👍  Evt. können wir dies noch direkt in die Entityklasse auslagern, damit wir den Controller schlank halten. Aber nicht so wichtig